hkmc2/shared/src/main/scala/hkmc2/codegen/HandlerLowering.scala

```diff
@@ -80,7 +80,7 @@ class HandlerLowering(using TL, Raise, Elaborator.State, Elaborator.Ctx):
   )
   
   object SimpleCall:
-    def apply(fun: Path, args: List[Path]) = Call(fun, args.map(Arg(false, _)))(true)
+    def apply(fun: Path, args: List[Path]) = Call(fun, args.map(Arg(false, _)))(true, false)
```

Comment:
This change looks dangerous. It seems to me that `SimpleCall` is a fairly generic helper, notably because of its name (in fact, we could probably move it to a place where more code could use it). Assuming such calls will always be effect-free may cause bugs to future users of the helper. It would be better to pass `isEffectful` as an argument to `apply`.

Reply:
So what should probably be documented for `isEffectful` is that currently it is `true` only when it is both an effectful call and the check is not added for the call (i.e. for tail call, it can be converted to isEffectful=false without any instrumentation as done currently in this PR)

But I don't know whether this is actually a good idea, because the handler lowering should only be ever done once (due to generation of a lot of additional class) and the parameter become a bit misleading if there were any passes after handler lowering that wish to know what calls can be effectful. I'm debating whether to change `isEffectful` to mean it is an effectful call (ignoring whether or not checks are added for the function)

I did find a potential future bug when changing the name tho.

Reply:
So you're saying you want to add a flag to the `Call` IR node although that flag only makes sense in the context of one specific transformation? Is there not a cleaner way?

At the very least it should have a much more specific and targeted name. Or, as you say, if that's not too much work you could generalize its meaning so it's not so misleading and could be used in other passes.

> I did find a bug when changing the name tho. There is now a usage of `super` which can be effectful and not instrumented (which actually exist already before this PR).

Curious what bug you are referring to. And is it fixed now?

Reply:
Wow ok, thanks for the explanation. That *is* a weird case.  We may want to just forbid it and throw an exception, though supporting the "obvious" semantics would be better. This can be done in a follow-up PR.